includes/class-amp-reader-themes.php

```diff
@@ -0,0 +1,313 @@
+		 *         @type string         $download_link  A link to the theme's zip file. If empty, the plugin will attempt to download the theme from wordpress.org.
+		 *     }
+		 * ]
+		 */
+		$themes = array_filter( (array) apply_filters( 'amp_reader_themes', $themes ) );
+
+		$themes = array_filter(
+			$themes,
+			function( $theme ) {
+				return is_array( $theme ) && ! empty( $theme ) && ! empty( $theme['screenshot_url'] ); // Screenshots are required.
+			}
+		);
+
+		$themes = array_map(
+			function ( $theme ) {
+				$theme['availability'] = $this->get_theme_availability( $theme );
+				return $theme;
+			},
+			$themes
+		);
+
+		$this->themes = $themes;
+
+		return $this->themes;
+	}
+
+	/**
+	 * Gets a reader theme by slug.
+	 *
+	 * @param string $slug Theme slug.
+	 * @return array|false Theme data or false if the theme is not found.
+	 */
+	public function get_reader_theme_by_slug( $slug ) {
+		return current(
+			array_filter(
+				$this->get_themes(),
+				static function ( $theme ) use ( $slug ) {
+					return $theme['slug'] === $slug;
+				}
+			)
+		);
+	}
+
+	/**
+	 * Retrieves theme data.
+	 *
+	 * @return array Theme data from the wordpress.org API.
+	 */
+	public function get_default_reader_themes() {
+		if ( null !== $this->default_reader_themes ) {
+			return $this->default_reader_themes;
+		}
+
+		$cache_key = 'amp_themes_wporg';
+		$response  = get_transient( $cache_key );
+		if ( ! $response ) {
+			// Note: This can be used to refresh the hardcoded raw theme data.
+			require_once ABSPATH . 'wp-admin/includes/theme.php';
+
+			$response = themes_api(
+				'query_themes',
+				[
+					'author'   => 'wordpressdotorg',
+					'per_page' => 24, // There are only 12 as of 05/2020.
+				]
+			);
+
+			if ( ! is_wp_error( $response ) ) {
+				set_transient( $cache_key, $response, DAY_IN_SECONDS );
+			}
+		}
+
+		if ( is_wp_error( $response ) ) {
+			return [ $this->get_classic_mode() ];
+		}
+
+		if ( is_array( $response ) ) {
+			$response = (object) $response;
+		}
+
+		$supported_themes = array_diff(
+			AMP_Core_Theme_Sanitizer::get_supported_themes(),
+			[ 'twentyten' ] // Because it is not responsive.
+		);
+
+		// Get the subset of themes.
+		$reader_themes = array_filter(
+			$response->themes,
+			static function ( $theme ) use ( $supported_themes ) {
+				return in_array( $theme->slug, $supported_themes, true );
+			}
+		);
+
+		$keys = [
+			'name',
+			'slug',
+			'preview_url',
+			'screenshot_url',
+			'homepage',
+			'description',
+			'requires',
+			'requires_php',
+		];
+
+		// Supply the screenshots.
+		$reader_themes = array_map(
+			static function ( $theme ) use ( $keys ) {
+				return array_merge(
+					array_reduce( // Provide empty defaults to make sure all keys are present.
+						$keys,
+						function( $result, $key ) {
+							$result[ $key ] = in_array( $key, [ 'requires', 'requires_php' ], true ) ? false : '';
+							return $result;
+						},
+						[]
+					),
```

Comment:
Do the keys `requires` and `requires_php` have to be false? If not:

```suggestion
					array_fill_keys( $keys, '' ), // Provide empty defaults to make sure all keys are present.
```

Reply:
Nice, I knew there must be something like this. Much more readable.